src/machine-config.h

```diff
@@ -122,6 +125,45 @@ struct htif_config final {
     bool yield_automatic{true};       ///< Make yield automatic available?
 };
 
+/// \brief VirtIO console device state config
+struct virtio_console_config final {};
+
+/// \brief VirtIO Plan 9 filesystem device state config
+struct virtio_p9fs_config final {
+    std::string tag{};            ///< Guest mount tag
+    std::string host_directory{}; ///< Path to the host shared directory
+};
+
+/// \brief VirtIO host forward state config
+struct virtio_hostfwd_config final {
+    bool is_udp{false};
```

Comment:
Is there any chance there would be another protocol ever? Say, unix? Also, what interface are you binding to on the host?

Reply:
> What interface are you binding to on the host.
`libslirp` has no option to choose an interface, I guess this can be automatically picked by the host address the machine is binding, I am thinking in binding always to loopback, so `lo` interface, do you agree?

Reply:
> Is there any chance there would be another protocol ever?
No, libslirp support only forwarding TCP or UDP.

With my recent changes it will bind to 127.0.0.1, therefore the loopback interface.

Reply:
I meant that, usually, people let you pick the interface to bind by specifying the associated ip address.

Reply:
I changed a bit how port forwarding is done, now it is the `--port-forward` or `-p` option, it has a syntax similar to docker `-p` option and now supports setting the host address to bind to.